File: cs50x/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_towers(int height);

int main(void)
{
    int height = get_height();
    print_towers(height);
}

int get_height(void)
{
    int height;

    do
    {
        height = get_int("How high do you want to make Mario's obstacle? Valid input is [1-8]? ");
    }
    while (height < 1 || height > 8);

    return height;
}

void print_towers(int height)
{
    // total possible width of the pattern is 2 times the
    // height plus two empty spaces between the towers
    int max_width = (2 * height - 2);

    for (int row_width = 0; row_width < height; row_width++)
    {
        // startspace is calculated by the max width, minus the 2 center spaces
        // and 2 times the height.

        for (int startspace = ((max_width - (2 * row_width - 2)) / 2); startspace > 1; startspace--)
        {
            // printf("%i", startspace);
            printf(" "); // Print the empty spaces before the blocks
        }

        for (int count = 0; count <= row_width; count++)
        {
            printf("#"); // Print the first set of blocks (number is same as height)
        }

        printf("  "); //  Print the two empty spaces in the middle

        for (int count = 0; count <= row_width; count++)
        {
            printf("#"); // Print the last set of blocks (number is same as height)
        }

        printf("\n"); // Print a newline to start the next row
    }
}

File: cs50x/sentimental-mario-less/mario.py
from cs50 import get_int

# Ask user how high of a pyrimad
while True:
    height = get_int("Height: ")

    # Check to see if height is between 1 and 8
    if height > 0 and height <= 8:
        break

# Check for valid input
if height < 1 or height > 8:
    print("Expected integer")
    exit(1)

else:
    # Increment the rows to the next row
    for row in range(height):

        # Print the empty blocks before the #
        # Total number is height - row #
        for space in range(height - (row + 1)):
            print(" ", end="")

        # Print the full blocks
        # Total number is the row #
        for block in range(row + 1):
            print("#", end="")

        print()
